pkg/controller/operators/catalog/operator.go

```diff
@@ -1180,6 +1186,77 @@ func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.In
 	return unpacked, out, nil
 }
 
+// gcInstallPlans garbage collects installplans that are too old
+// installplans are ownerrefd to all subscription inputs, so they will not otherwise
+// be GCd unless all inputs have been deleted.
+func (o *Operator) gcInstallPlans(log logrus.FieldLogger, namespace string) {
+	allIps, err := o.lister.OperatorsV1alpha1().InstallPlanLister().InstallPlans(namespace).List(labels.Everything())
+	if err != nil {
+		log.Warn("unable to list installplans for GC")
+	}
+
+	if len(allIps) <= maxInstallPlanCount {
+		return
+	}
+
+	// we only consider maxDeletesPerSweep more than the allowed number of installplans for delete at one time
+	ips := allIps
+	if len(ips) > maxInstallPlanCount + maxDeletesPerSweep {
+		ips = allIps[:maxInstallPlanCount+maxDeletesPerSweep]
```

Comment:
This depends on the lister returning `InstallPlans` in descending `createdAt` order, which [doesn't look like the case](https://github.com/operator-framework/operator-lifecycle-manager/blob/cc2112c70498f7a08c84c8886b5c360657568de1/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go#L109).

Reply:
forgot this is being run multiple times; i.e. successive sweeps will clean these up.